addons/udes_common/models/fields.py

```diff
@@ -0,0 +1,59 @@
+import datetime as dt
+from odoo import fields
+
+
+DATE_LENGTH = fields.DATE_LENGTH
+DATETIME_FORMAT = fields.DATETIME_FORMAT + ".%f"
+DATETIME_LENGTH = len(dt.datetime.now().strftime(DATETIME_FORMAT))
+
+
+class PreciseDatetime(fields.Datetime):
+    """Store microsecond-precision timestamps."""
+
+    type = "precise_datetime"
+    column_type = ("timestamp", "timestamp")
+    column_cast_from = ("date",)
+
+    @staticmethod
+    def now(*args):
+        """ Return the current day and time in the format expected by the ORM.
+            This function may be used to compute default values.
+        """
+        return dt.datetime.now().strftime(DATETIME_FORMAT)
+
+    @staticmethod
+    def from_string(value):
+        """ Convert an ORM ``value`` into a :class:`datetime` value. """
+        if not value:
+            return None
+        value = value[:DATETIME_LENGTH]
+        if len(value) == DATE_LENGTH:
+            value += " 00:00:00.000000"
+        return dt.datetime.strptime(value, DATETIME_FORMAT)
+
+    @staticmethod
+    def to_string(value):
+        """ Convert a :class:`datetime` value into the format expected by the ORM. """
+        return value.strftime(DATETIME_FORMAT) if value else False
+
+    def convert_to_column(self, value, record, values=None):
+        return super(PreciseDatetime, self).convert_to_column(value or None, record, values)
+
+    def convert_to_cache(self, value, record, validate=True):
+        if not value:
+            return False
+        if isinstance(value, str):
+            if validate:
+                # force parsing for validation
+                self.from_string(value)
+            value = value[:DATETIME_LENGTH]
+            if len(value) == DATE_LENGTH:
+                value += " 00:00:00.000000"
+            return value
+        return self.to_string(value)
+
+    def convert_to_display_name(self, value, record):
```

Comment:
If this method were to strip the microseconds from the time, would this mean it could be rendered in the views? (Not sure we would ever want to show the microseconds in the UI.)

Reply:
Maybe i can try to use that method where i compute the datetime field which is shown in the UI.
The PreciseDatetime field was giving error on UI, thats why created another one datetime to show in UI

Reply:
So i was wondering if this method is the one that formats it for the UI, and if it is just a precision issue, in this return the striped down datetime then it can be viewed without an additional computed field.

Reply:
Tbh tried to show the precise datetime in the UI , and could not. Would appreciate if there is a suggestion how, and i can give a try. 
In that case yes the computed field should be removed, as the reason added was only to show in UI

Reply:
So just for visibility: We had a chat offline and really I thought we want the precise datetime in most places (like u_done_datetime etc) so the events have it but in the frontend everything gets trimmed to ignore microseconds. 

1. Handle datetime as a precise datetime everywhere- probably to risky for this story.
2. Set the same type as the datetime field, i.e type="datetime" so odoo front end renders it like datetime. But that does mean we have two feilds with the same type.
3. Add the new field properly, this involves files like basic_fields.js, field_registry etc. updating them for the new field. 

Only 2 is probably acceptable for this story. 

The main issue was raised is that if we want the precise datetime for events, and want to show any of those fields in the UI will need an additional computed field. This seems like over kill, 1 might be ok, but don't want to let it grow really.

I would prefer maybe an issue to either do 1 or 3, not sure any side affects of 2.

Reply:
Changed the type to datetime in PreciseDatetime class, assigned a picking to a user and when psql on database the datetime field shows with microseconds while in UI without and doesnt raise any error which i think is waht we need.

Reply:
@peterpratt13 As you say I think option 2 is best for this story unless we have a breakthrough for the other ones. I'll create an issue to handle precise date times nicely in the front end.